### PR TITLE
Document GHC-59840

### LINKED ISSUE #112: DsMultiplicityCoercionsNotSupported -> 59840

### PR DESCRIPTION
**Pull request by BinderDavid**
Fixes #112 

This one is for the connaisseurs :)

### CODE REVIEW

File: message-index/messages/GHC-59840/linearPolyType/index.md [hunk 1]
```diff
@@ -0,0 +1,14 @@
+---
+title: <insert a title for this example here>
```
Comment (david-christiansen) [inline message-index/messages/GHC-59840/linearPolyType/index.md:2]: This title seems uninformative... What about "Unsupported multiplicity equality in GADT" or something like that?

File: message-index/messages/GHC-59840/linearPolyType/after/LinearPolyType.hs [hunk 2]
```diff
@@ -0,0 +1,3 @@
+module LinearPolyType where
+
+-- This program can currently not be written :(
```
Comment (david-christiansen) [inline message-index/messages/GHC-59840/linearPolyType/after/LinearPolyType.hs:3]:
```suggestion
-- This program cannot currently be written
```

File: message-index/messages/GHC-59840/index.md [hunk 3]
```diff
@@ -0,0 +1,16 @@
...
+title: GHC does not support GADTs or type families which witness equality of multiplicities
+summary: GHC does not support GADTs or type families which witness equality of multiplicities
+severity: error
+introduced: 9.6.1
+---
+
+There are two features which GHC supports, and whose interaction is problematic and leads to this error:
+
+- Programming constructs which build upon the type equality constraint `a ~ b`, such as GADTs and type families.
+- Linear types, which supports functions like `a %m -> b`. In this function type `m` stands for a *multiplicity* which can be either `One`(also written `1`) or `Many`, depending on how often the argument to the function is used in the body of the function.
```
Comment (david-christiansen) [inline message-index/messages/GHC-59840/index.md:11]:
```suggestion
- Linear types, which supports functions like `a %m -> b`. In this function type `m` stands for a *multiplicity* which can be either `One`(also written `1`) or `Many`, which determines how often the argument to the function may be used in the body of the function.
```

**Review by david-christiansen — CHANGES_REQUESTED** [review]
This is a hairy one! Well done!

File: message-index/messages/GHC-59840/linearPolyType/index.md (hunk 1 above)
Comment (BinderDavid) [reply message-index/messages/GHC-59840/linearPolyType/index.md]: Oh 😅 . I updated the title of the example.

**Pull request merged**